src/_utils/focusVisible/FocusVisibleProvider.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState, ReactNode } from 'react'
+
+// A React hook based on: https://github.com/WICG/focus-visible
+
+interface FocusVisibleProviderProps {
+  children: ReactNode
+}
+
+export const FocusVisibleContext = React.createContext({
+  hadKeyboardEvent: true,
+  isInitialized: false,
+})
+
+export const FocusVisibleProvider = ({ children }: FocusVisibleProviderProps) => {
+  const [hadKeyboardEvent, setHadKeyboardEvent] = useState(true)
+
+  const onPointerDown = () => {
+    setHadKeyboardEvent(false)
+  }
+
+  /**
+   * When the polfyill first loads, assume the user is in keyboard modality.
+   * If any event is received from a pointing device (e.g. mouse, pointer,
+   * touch), turn off keyboard modality.
+   * This accounts for situations where focus enters the page from the URL bar.
+   * @param {PointerEvent} e
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onInitialPointerMove = (e: any) => {
```

Comment:
Don't find a better Typescript event definition for for now because of `target.nodeName`

Reply:
Try `MouseEvent`

Reply:
Doesn't work either 😕